src/state/utils/store_io.ts

```diff
@@ -38,19 +49,23 @@ export const serializedStoreDataToUrl = (data: SerializedStoreData): string => {
 // DESERIALIZATION ///////////////////////////////////////////////////////////////////////
 
 /**
- * Loads the viewer state from the given URL  parameters. Note that this MUST be
- * called after the dataset is loaded and set in the store.
+ * Loads the viewer state from the given URL parameters. Note that this MUST be
+ * called after the collection and dataset are loaded and set in the store.
  */
 export const loadViewerStateFromParams = async (store: Store<ViewerState>, params: URLSearchParams): Promise<void> => {
-  // NOTE: Ordering is important here, because of dependencies between slices.
   // 1. No dependencies:
-  loadDatasetSliceFromParams(store.getState(), params);
   loadConfigSliceFromParams(store.getState(), params);
 
-  // 2. Dependent on dataset fields:
-  loadTimeSliceFromParams(store.getState(), params);
+  // 2. Dependent on dataset object:
+  loadBackdropSliceFromParams(store.getState(), params);
+  loadDatasetSliceFromParams(store.getState(), params);
+  loadScatterPlotSliceFromParams(store.getState(), params);
+  loadThresholdSliceFromParams(store.getState(), params);
```

Comment:
nit: would these be more specific if they passed in their relevant slices instead of the whole store?

Reply:
Each of the slice loaders is generally parameterized by only that slice, with some exceptions for slices that need access to the dataset or other information. I wouldn't really be able to get only the individual slice contents unless I wrote selectors for each, which would be a big headache. 

The slices really only exist to separate out logic for readability, but they're all sort of mixed together into a big state soup in the actual store.